debian/control

```diff
@@ -17,10 +17,10 @@ Description: Tvheadend
    - supports a variety of inputs (DVB-S(2)/T(2)/C, ISDB-S/T/C, ATSC-T/C, IPTV, SAT>IP, HDHR)
    - supports a variety of clients (Movian, Smart TV, Kodi, VLC)
 
-Package: tvheadend-dbg
+Package: tvheadend-dbg-armv6l
 Architecture: any
 Section: debug
 Priority: extra
-Depends: tvheadend (= ${binary:Version}), ${misc:Depends}
+Depends: tvheadend-armv6l (= ${binary:Version}), ${misc:Depends}
```

Comment:
While I am unsure about the purpose of changing the package name, which breaks things for all admins who currently have the `tvheadend` package installed, at least I would suggest:
```sh
Provides: tvheadend
Breaks: tvheadend
Replaces: tvheadend
```
to allow a seamless migration from one to the other package. But what is the purpose of renaming the package in the first place?

Reply:
There is no point in migrating as only tvheadend-armv6l will work on the armv6l architecture.

I am not sure what other options there are. Currently the armv6l and armhf packages are colliding as they are both armhf as far as dpkg is concerned. What other options are there?

Reply:
But it used to work in the past, didn't it? Or did those packages never work?

The solution is basically already there, as those packages are (supposed to be) located in another APT repository:
- ARMv7: https://dl.cloudsmith.io/public/tvheadend/tvheadend/deb/debian
- ARMv6: https://dl.cloudsmith.io/public/tvheadend/tvheadend/deb/raspbian

So as long as the ARMv6 ones are uploaded to `deb/raspbian`, there is no collision.

As far as I could see, the issue is actually wrong compiling flags used during FFmpeg related build stages, and they are still there: https://github.com/tvheadend/tvheadend/actions/runs/8208034460/job/22450593063
```
gcc -Wno-maybe-uninitialized -Wshadow -O3 -ffast-math  -Wall -I. -I. -fPIE -std=gnu99 -D_GNU_SOURCE -mcpu=cortex-a8 -mfpu=neon
```
`-mfpu=neon` means ARMv7-only. For ARMv6, this needs to be `-mfpu=vfp`. And later an `armv7l-...` host triple is used. Renaming the package does not solve incompatible build flags, of course 😉.

The problem with `dpkg` and `armhf` is the reason why Debian does not/cannot support those old Raspberry Pi models, and why Raspbian exists, as a dedicated Debian clone with ARMv6-compatible `armhf` arch. The Raspbian debhelper (which you already used, AFAICS) should however handle this correctly.

I'll quickly check back, but I think that `--arch=armel` is used here might be even another problem, since this is not the native `dpkg` architecture used on RPi. As packages were installed, it might be ignored/overridden: https://github.com/tvheadend/tvheadend/blob/master/Autobuild/armv6l.sh#L1

Reply:
> But it used to work in the past, didn't it? Or did those packages never work?

If they did then it was pure coincidence, the armv6l and armv7 packages share the same name, so whichever one gets uploaded last was available for downloading.

> The solution is basically already there, as those packages are (supposed to be) located in another APT repository:

There are more differences between raspbian and debian though, so we are maintaining builds for raspbian built on armv6l and armv7. You can check the build actions here on GitHub to get a better understanding of what is built and how it is built.

> `-mfpu=neon` means ARMv7-only. For ARMv6, this needs to be `-mfpu=vfp`. And later an `armv7l-...` host triple is used.

Why does ffmpeg use invalid build flags though? It is running on a armv6l-container, so it should build something that is compatible with this container as we are not cross-compiling. That would mean self-compiling is also causing issues. Those issues might never appear though as the code might never get executed. However, someone should do further testing there to confirm if it's really an issue.

> Renaming the package does not solve incompatible build flags, of course 😉.

It's a way to ensure that the changes to actually matter though. If the other builds always overwrite these changes/packages or are competing then any changes would be useless/unreliable.

Reply:
Thinking more about it: Do I understand it correctly that you suggest kicking out the armv7-builds for raspbian so it only runs armv6-builds and armv7-users should switch to the debian repo then?

Reply:
> There are more differences between raspbian and debian though, so we are maintaining builds for raspbian built on armv6l and armv7. You can check the build actions here on GitHub to get a better understanding of what is built and how it is built.

I tried to understand it, but failed to be certain. However, I did see for a short time the `tvheadend-armv6l` packages on the `deb/debian` repo as well, hence that indeed both arch builds are uploaded to both repos. Maybe it is better to address this part instead, i.e. trying to use `debian` and `raspbian` more like they are natively intended to be used:
- Generally, all ARMv7/8-capable RPi models can use `deb/debian`. Raspbian is a Debain clone, so there are/should be library/dependency issues when doing so. All RPi-specific builds, for hardware acceleration etc, are shipped not by Raspbian, but by `archive.raspberrypi.org`, the dedicated RPi repository, which can be used on Debian and Raspbian likewise.
- And the other way round, ARMv7/8-capable RPi models should have no problems using ARMv6 builds.
- So not sure which other issues you faced, when using `deb/raspbian` for `armv6l` only, and `deb/debian` for `armv7l` only, but maybe they can be solved.

> Why does ffmpeg use invalid build flags though? It is running on a armv6l-container, so it should build something that is compatible with this container as we are not cross-compiling. That would mean self-compiling is also causing issues. Those issues might never appear though as the code might never get executed. However, someone should do further testing there to confirm if it's really an issue.

I was wondering about the same: I see you use a Raspbian container, hence correct toolchain etc assured, and you just use debhelper (right?). At some point, the parental build flags seem to not be passed through, but not sure why. What has to be kept in mind is, that QEMU cannot emulate ARMv6, but only ARMv7, i.e. when you run `uname -m` within the emulated container, you'll see `armv7l`, regardless whether it is a Debian or Raspbian userland. And many tools derive the target architecture from `uname -m` (or more direct calls), hence it is not rare that one needs to declare ARMv6 target/flags more explicitly in these cases.

Maybe we should continue the conversation in #1665. I'll try to have a closer look that the build/scripts.

Reply:
> I tried to understand it, but failed to be certain. However, I did see for a short time the `tvheadend-armv6l` packages on the `deb/debian` repo as well, hence that indeed both arch builds are uploaded to both repos.

That was a mistake as I accidentally renamed *ALL* packages.... The build that is running now should be correct.